### PR TITLE
ref(sdk-updates-broadcast): Update design & add raven alert

### PR DESCRIPTION
**Pull request by priscilawebdev**
INGEST-158

https://getsentry.atlassian.net/secure/RapidBoard.jspa?rapidView=113&projectKey=INGEST&view=planning&selectedIssue=INGEST-158&quickFilter=196&issueLimit=100

https://www.notion.so/sentry/Update-your-SDKs-notes-9ee74ad6566e4ec99bf10b01b7009981

**Preview:**

image

- [x] Add warning alert
- [x] Add a deprecated tag to raven projects only
- [x] Migrate link goes to corresponding SDK docs migration page -> added some links, but unfortunately, they don't cover all deprecated sdks that we have
- [ ] Update link goes to corresponding SDK docs installation page -> we decided to discuss this further internally
- [x] Sort by deprecated, then by platform
- [x] Updates should only be shown to users who are project members or users who have open membership or org write permission

ps: will test this on staging

### CODE REVIEW

**Comment by lobsterkatie** [comment]
This looks great!

Two questions:

1) Would a user really get multiple alerts for the same project? If so, how/why? IMHO this shouldn't happen.

2) Can we create a map of internal SDK name (`sentry.javascript.browser`) to external SDK name (`@sentry/browser`), so that we can display the latter rather than the former?

Also, a stretch goal to think about:

IDK if this applies to any raven SDKs other than `raven-js` (maybe the old PHP SDK?), but at least for `raven-js`, it'd be great if we could mimic what Markus's original in-event banner did and look for indicators that the user is using a framework we specifically support, so that we can tailor our suggestions appropriately. For example, if someone is using `raven-js` and we can tell from events they send in that they're also using react, we could suggest `@sentry/react` rather than `@sentry/browser`.

File: src/sentry/sdk_updates.py [hunk 1]
```diff
@@ -86,11 +86,17 @@ def __init__(self, sdk_name, new_sdk_version, ignore_patch_version):
         self.ignore_patch_version = ignore_patch_version
 
     def to_json(self):
+        sdkUrls = (
+            settings.SDK_MIGRATION_URLS
+            if dict(settings.SDK_MIGRATION_URLS).get(self.sdk_name)
```
Comment (untitaker) [inline src/sentry/sdk_updates.py:91]:
```suggestion
            if settings.SDK_MIGRATION_URLS.get(self.sdk_name)
```

File: src/sentry/sdk_updates.py [hunk 2]
```diff
@@ -86,11 +86,17 @@ def __init__(self, sdk_name, new_sdk_version, ignore_patch_version):
         self.ignore_patch_version = ignore_patch_version
 
     def to_json(self):
+        sdkUrls = (
```
Comment (untitaker) [inline src/sentry/sdk_updates.py:89]: snake_case!

File: src/sentry/conf/server.py [hunk 3]
```diff
@@ -1937,6 +1937,19 @@ def build_cdc_postgres_init_db_volume(settings):
     "sentry-swift": "https://docs.sentry.io/clients/cocoa/",
 }
 
+# Some of the links below are not ideal, but that is all migration documentation we currently have and can provide at this point
+SDK_MIGRATION_URLS = {
```
Comment (untitaker) [inline src/sentry/conf/server.py:1941]: what's the point of defining this as a separate dictionary? The way you wrote the code, doesn't this mean you will never link to any `/clients/foo/` link if there's a migration guide, right?

**Review by untitaker — COMMENTED** [review]
not sure if this achieves the desired outcome

File: src/sentry/conf/server.py (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/conf/server.py:1941]: I just wanted to keep it separated but you're right

File: static/app/components/sidebar/broadcastSdkUpdates.tsx [hunk 4]
```diff
@@ -26,18 +32,83 @@ const flattenSuggestions = (list: ProjectSdkUpdates[]) =>
...
+
+        if (!project) {
+          return null;
+        }
+
+        // Updates should only be shown to users who are project members or users who have open membership or org write permission
+        const hasPermissionToSeeUpdates =
+          project.isMember ||
+          organization.features.includes('open-membership') ||
+          organization.features.includes('org:write');
```
Comment (markstory) [inline static/app/components/sidebar/broadcastSdkUpdates.tsx:61]: features will never include `org:write`. I think you want `organization.access` here instead.

File: static/app/utils/getSdkUpdateSuggestion.tsx [hunk 5]
```diff
@@ -20,29 +22,39 @@ function getSdkUpdateSuggestion({
...
+          ? tct('Update to [sdk-name]@v[new-sdk-version]', {
+              ['sdk-name']: sdkName,
               ['new-sdk-version']: newSdkVersion,
             })
+          : tct(
+              'Update your SDK from [sdk-name]@v[sdk-version] to [sdk-name]@v[new-sdk-version]',
+              {
+                ['sdk-name']: sdkName,
+                ['sdk-version']: sdk.version,
+                ['new-sdk-version']: newSdkVersion,
```
Comment (markstory) [inline static/app/utils/getSdkUpdateSuggestion.tsx:38]: Do you need the `[]` if you're also using string keys?

**Review by markstory — APPROVED** [review]
Looks good for staging.

File: static/app/components/sidebar/broadcastSdkUpdates.tsx (hunk 4 above)
Comment (priscilawebdev) [reply static/app/components/sidebar/broadcastSdkUpdates.tsx:61]: oh good catch!! thanks

File: static/app/components/sidebar/broadcastSdkUpdates.tsx [hunk 6]
```diff
@@ -47,50 +118,31 @@ const BroadcastSdkUpdates = ({projects, sdkUpdates}: Props) => {
         'We recommend updating the following SDKs to make sure you’re getting all the data you need.'
       )}
     >
+      {!!deprecatedRavenSdkUpdates.length && (
+        <StyledAlert type="warning" icon={<IconWarning />}>
+          {tct(
+            `[first-sentence]. Any SDK that has the package name ‘raven’ may be missing data. Migrate to the latest SDK version.`,
+            {
+              ['first-sentence']: tn(
+                'You have %s project using a deprecated version of the Sentry client',
```
Comment (scefali) [inline static/app/components/sidebar/broadcastSdkUpdates.tsx:127]: @priscilawebdev I would put the `.` as part of the translation so it translates the full sentence.

**Review by scefali — APPROVED** [review]

**Pull request merged**